src/ansys/fluent/core/services/field_data.py

```diff
@@ -1285,3 +1399,39 @@ def get_pathlines_field_data(
                 field_name: pathlines_data[surface_ids[count]][field_name],
             }
         return path_lines_dict
+
+    def get_mesh(self, zone_id: int) -> Mesh:
+        """Get mesh for a zone.
+
+        Parameters
+        ----------
+        zone_id : int
+            Zone ID.
+
+        Returns
+        -------
+        Mesh
+            Mesh object containing nodes and elements.
+        """
+        request = FieldDataProtoModule.GetSolverMeshNodesRequest(
+            domain_id=1, thread_id=zone_id
+        )
+        response = self._service.get_solver_mesh_nodes(request)
+        nodes = response.nodes
+        nodes = [Node(_id=node.id, x=node.x, y=node.y, z=node.z) for node in nodes]
+        nodes = sorted(nodes, key=lambda x: x._id)
+        request = FieldDataProtoModule.GetSolverMeshElementsRequest(
+            domain_id=1, thread_id=zone_id
+        )
+        response = self._service.get_solver_mesh_elements(request)
+        elements = response.elements
+        elements = [
+            Element(
+                _id=element.id,
+                element_type=CellElementType(element.element_type),
+                node_indices=[(id - 1) for id in element.node_ids],
+            )
+            for element in elements
+        ]
+        elements = sorted(elements, key=lambda x: x._id)
```

Comment:
TODO: It would be more performant to do the 0-based indexing and sorting in the server-side. Maybe we can eliminate ids at the protobuf layer also.